### DISCUSSION TITLE
How to implement ARjs with react?

### DISCUSSION THREAD

**Discussion by ivanlen** [AR-js-org/AR.js#433] [Q&A]
Hey, I am learning react and I would like to implement ARjs in react.
Is there some kind of working example on how to implement the code with react?
I am trying to reproduce the [location based example](https://github.com/AR-js-org/AR.js/#location-based-example) but to be honest I am kind of lost.

What I am expecting is building a component that based on some logic returns/renders the example.

```js
export default function TestLoad() {
    const router = useRouter();
    const query = router.query;
    console.log(query.info)

    if (query.info == 'true') {
        console.log(__html.default)
        console.log('TRUE RETURN')
        return (
            // CODE HERE
        )
}
```

I tried using `dangerouslySetInnerHTML` but is not working. I guess that I have to use the [npm package](https://www.npmjs.com/package/@ar-js-org/ar.js), but I don't know how to implement it with react.

I managed to modify the  [location based example](https://github.com/AR-js-org/AR.js/#location-based-example) with gltf files that I made with blender and everything is working great in a html file.

So the question would be how to migrate my html code to a code that can be implemented in react.

**Comment by kunal-rp** [comment]
You will have to reference the exports here:`ar.js/three.js/build/ar-threex`. There you can import in the context and toolkit api's required to instantiate the feed and install the ar controllers. From there you will need to use state management to pass the context s.t. you can get the marker positions.

From there you can add the three.js objects to display. Since you are using reach, you might as well use react-fiber

**Reply by ivanlen** [reply]
Thank you very much for your answer. I am still kind of noob with react. I can follow the overall idea, but I am not sure I can implement it.
If its not too much, can you provide some kind of code snippet so then I can adapt it?

**Reply by kalwalt** [reply]
You can try the https://github.com/artcom/react-three-arjs by @j-era or do you want to develop your own package?

**Reply by ivanlen** [reply]
Thanks @kalwalt . No, ideally I don't want to develop my own package. I checked out the examples and it doesn't seem to have support for location based ar.

**Reply by kalwalt** [reply]
Yes you are right for now it doesn't support location based ar.

**Comment by nickw1** [comment]
@ivanlen feel free to submit an issue to the author of `react-three-arjs`, or even an issue here. We may be able to develop a React package for location-based AR, time permitting.

**Comment by sdmrf** [comment]
I have successfully integrated AR.js with React.js; however, I am uncertain whether I have followed the correct approach. I would greatly appreciate it if someone could assist me in verifying the integration. I am relatively new to this and dedicated three days to unraveling the process, eventually achieving the integration. Your help would be invaluable.

**Reply by ivanlen** [reply]
any link?

**Reply by sdmrf** [reply]
I have been conducting experiments on it, and it is still a work in progress. You are welcome to explore the demo at the following link: https://sdmrf.me/ap. Additionally, you can review the source code on my GitHub repository: https://github.com/sdmrf/ap. Your feedback would be greatly appreciated, as it will aid in it's improvement.

![demo_img](https://github.com/AR-js-org/AR.js/assets/79003834/fc615408-3ddf-457a-9774-b7217fe3c619)

**Comment by Rossella-Mascia-Neosyn** [comment]
@sdmrf Your link no longer works, do you have another project?
